### PR TITLE
Added the agent id as header in every request

### PR DESCRIPTION
**Pull request by talgoldy**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 ### Fixed
 - Default versions reported in `AgentId` now automatically generated and match `pom.xml`.
 - Method `setEnablePatterns` now works properly. [Trello 1714](https://trello.com/c/jQgW5dpz)
+### Updated
+- Adding agent id all requests headers. [Trello 1697](https://trello.com/c/CzhUxOqE)
```
Comment (itaibh) [inline CHANGELOG.md:6]: - Adding agent id **to** all requests headers. [Trello 1697](https://trello.com/c/CzhUxOqE)

File: eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public EyesBase() {
 
         initProviders();
 
-        setServerConnector(new ServerConnector());
+        setServerConnector(new ServerConnector(getBaseAgentId()));
```
Comment (itaibh) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java:85]: Why `getBaseAgentId()` and not `getFullAgentId()`?

File: eyes.connectivity.java.jersey2x/src/main/java/com/applitools/eyes/ServerConnector.java [hunk 3]
```diff
@@ -50,27 +48,27 @@
...
+        super(logger, serverUrl, TIMEOUT, agentId);
         endPoint = endPoint.path(API_PATH);
     }
 
     /***
      * @param logger A logger instance.
      */
-    public ServerConnector(Logger logger) {
-        this(logger, GeneralUtils.geServerUrl());
+    public ServerConnector(Logger logger, String agentId) {
```
Comment (itaibh) [inline eyes.connectivity.java.jersey2x/src/main/java/com/applitools/eyes/ServerConnector.java:59]: In addition to this, I would keep the older implementation and pass a constant `null` so you won't have to change the tests.

Comment (talgoldy) [reply eyes.connectivity.java.jersey2x/src/main/java/com/applitools/eyes/ServerConnector.java:59]: But if we have here 4 constructors it means there will be 8 now. Or you mean only this specific one?

Comment (itaibh) [reply eyes.connectivity.java.jersey2x/src/main/java/com/applitools/eyes/ServerConnector.java:59]: no, just to this one. This is the one used in the tests.

Comment (itaibh) [reply eyes.connectivity.java.jersey2x/src/main/java/com/applitools/eyes/ServerConnector.java:59]: Also, make sure you actually use need all those ctors.

File: eyes.connectivity.java.jersey1x/src/main/java/com/applitools/eyes/ServerConnector.java [hunk 4]
```diff
@@ -550,7 +563,10 @@ public IPutFuture renderPutResource(final RunningRender runningRender, final RGr
         } else {
             builder = target.entity(content, MediaType.APPLICATION_OCTET_STREAM_TYPE);
         }
-        builder = builder.header("X-Auth-Token", renderingInfo.getAccessToken());
+        builder = builder
+                .header("X-Auth-Token", renderingInfo.getAccessToken())
+                .header(AGENT_ID_CUSTOM_HEADER, agentId)
+                .header(AGENT_ID_CUSTOM_HEADER, agentId);
```
[the remark this answers is not in the document]
Comment (talgoldy) [reply eyes.connectivity.java.jersey1x/src/main/java/com/applitools/eyes/ServerConnector.java:569]: yea

File: eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java [hunk 5]
```diff
@@ -143,6 +143,18 @@ public void setServerConnector(IServerConnector serverConnector) {
...
+            return null;
+        }
+
+        if (serverConnector.getAgentId() == null) {
+            serverConnector.setAgentId(getFullAgentId());
+        }
+
+        return serverConnector;
+    }
+
```
[the remark this answers is not in the document]
Comment (talgoldy) [reply eyes.sdk.core/src/main/java/com/applitools/eyes/EyesBase.java:157]: ok

**Pull request closed without merging**